src/utils.ml

```diff
@@ -205,6 +205,33 @@ module Ast_io = struct
         output_string oc Input_version.Ast.Config.ast_impl_magic_number;
         output_value oc input_name;
         output_value oc st
+
+    module Read_bin = struct
+    type ast =
+    | Intf of signature
+    | Impl of structure
+
+    type read_error' = Read_error of string
```

Comment:
Why do you prefer to define a new type to encapsulate a string instead of using a string directly?